tools/buildHelper/buildHelper.psm1

```diff
@@ -721,13 +721,80 @@ function Get-TestParams
+    {
+        switch ($dockerFileName)
+        {
+            {($_ -like "alpine") -or ($_ -like "mariner")}
+            {
+                # $pwshInstallerFileUri = "https://github.com/PowerShell/PowerShell/releases/download/v$($psversion)/powershell-$($psversion)-linux-x64.tar.gz"
+                $pwshInstallerFileName = "powershell-$($psversion)-linux-x64.tar.gz"
+            }
+            {($_ -like "debian") -or ($_ -like "ubuntu")}
+            {
+                # $pwshInstallerFileUri = "https://github.com/PowerShell/PowerShell/releases/download/v$($psversion)/powershell_$($psversion)-1.deb_amd64.deb"
+                $pwshInstallerFileName = "powershell_$($psversion)-1.deb_amd64.deb"
+            }
+            {($_ -like "nanoserver") -or ($_ -like "windowsserver")}
+            {
+                # $pwshInstallerFileUri = "https://github.com/PowerShell/PowerShell/releases/download/v$($psversion)/PowerShell-$($psversion)-win-x64.zip"
+                $pwshInstallerFileName = "PowerShell-$($psversion)-win-x64.zip"
+            }
+            "ubi"
+            {
+                # $pwshInstallerFileUri = "https://github.com/PowerShell/PowerShell/releases/download/v$($psversion)/powershell-$($psversion)-1.rh.x86_64.rpm"
+                $pwshInstallerFileName = "powershell-$($psversion)-1.rh.x86_64.rpm"
+            }
+        }
+    }
+    elseif($channelTag -eq "preview")
+    {
+        switch ($dockerFileName)
+        {
+            {($_ -like "alpine") -or ($_ -like "mariner")}
+            {
+                # $psversion will be like 7.3.0-preview.1
+                # $packageversion will be like 7.3.0_preview.1
+                # $pwshInstallerFileUri = "https://github.com/PowerShell/PowerShell/releases/download/v$($psversion)/powershell-$($psversion)-linux-x64.tar.gz"
+                $pwshInstallerFileName = "powershell-$($psversion)-linux-x64.tar.gz"
+            }
+            {($_ -like "debian") -or ($_ -like "ubuntu")}
+            {
+                # $pwshInstallerFileUri = "https://github.com/PowerShell/PowerShell/releases/download/v$($psversion)/powershell-preview_$($psversion)-1.deb_amd64.deb"
+                $pwshInstallerFileName = "powershell-preview_$($psversion)-1.deb_amd64.deb"
+            }
+            {($_ -like "nanoserver") -or ($_ -like "windowsserver")}
+            {
+                # $pwshInstallerFileUri = "https://github.com/PowerShell/PowerShell/releases/download/v$($psversion)/PowerShell-$($psversion)-win-x64.zip"                
+                $pwshInstallerFileName = "PowerShell-$($psversion)-win-x64.zip"
+            }
+            "ubi"
+            {
+                # $pwshInstallerFileUri = "https://github.com/PowerShell/PowerShell/releases/download/v$($psversion)/powershell-preview-$($psversion)-1.rh.x86_64.rpm"
+                $pwshInstallerFileName = "powershell-preview-$($psversion)-1.rh.x86_64.rpm"
+            }
+        }
+    }
```

Comment:
We already calculate the package name on lines 808 and 832 (in your version).
Your version hard codes the package format which is incorrect.  Meta already has this information, so we don't have to change the code when the package format changes.

Also, Mariner uses tar.gz for stable and rpm for preview (currently.)

Reply:
ah good point, I'll pull the package name from the meta. Then I also don't need to populate that based on OS, it can take what we specify in the meta

Reply:
Why do we need a new property?